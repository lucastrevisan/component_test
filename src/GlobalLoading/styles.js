import styled, { keyframes } from "styled-components";
import { rem } from "../Common/styles/tools";
import { darkBlue, blue10, white } from "../Common/styles/settings";

const loading = keyframes`
  from {
    left: 50%;
    width: 0;
    z-index: 100;
  }

  33.333% {
    left: 0;
    width: 100%;
    z-index: 10;
  }

  to {
    left: 0;
    width: 100%;
  }
`;

export const Wrapper = styled.div`
  position: fixed;
  top: 0;
  width: 100%;
  height: ${rem(3)};
  background-color: ${white};
  z-index: 100000;
`;

export const Bar = styled.div`
  content: "";
  display: inline-block;
  position: absolute;
  width: 0;
  height: 100%;
  left: 50%;
  text-align: center;

  :nth-child(1) {
    background-color: ${darkBlue};
    animation: ${loading} 3s linear infinite;
  }

  :nth-child(2) {
    background-color: ${blue10};
    animation: ${loading} 3s linear 1s infinite;
  }

  :nth-child(3) {
    background-color: ${white};
    animation: ${loading} 3s linear 2s infinite;
  }
`;
