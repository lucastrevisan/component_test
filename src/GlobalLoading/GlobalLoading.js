import React from "react";
import { bool } from "prop-types";
import { Wrapper, Bar } from "./styles";

function GlobalLoading({ loading }) {
  if (loading) {
    return (
      <Wrapper data-test="Loading">
        <Bar />
        <Bar />
        <Bar />
      </Wrapper>
    );
  }

  return null;
}

GlobalLoading.defaultProps = {
  loading: null
};

GlobalLoading.propTypes = {
  loading: bool
};

export default GlobalLoading;
