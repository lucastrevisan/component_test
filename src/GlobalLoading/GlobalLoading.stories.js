import React from "react";
import { storiesOf } from "@storybook/react"; // eslint-disable-line
import { withInfo } from "@storybook/addon-info";

import GlobalLoading from "./GlobalLoading";

storiesOf("GlobalLoading", module).add(
  "default",
  withInfo()(() => <GlobalLoading loading={true} />)
);
