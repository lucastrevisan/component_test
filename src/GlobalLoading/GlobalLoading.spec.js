import React from "react";
import { render } from "enzyme";

import GlobalLoading from "./GlobalLoading";

describe("GlobalLoading component", () => {
  it("should match snapshot", () => {
    expect(render(<GlobalLoading />)).toMatchSnapshot();
  });

  it("should match snapshot with an opened GlobalLoading", () => {
    expect(render(<GlobalLoading loading={true} />)).toMatchSnapshot();
  });
});
