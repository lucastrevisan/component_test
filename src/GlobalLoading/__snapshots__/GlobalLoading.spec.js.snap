// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalLoading component should match snapshot 1`] = `null`;

exports[`GlobalLoading component should match snapshot with an opened GlobalLoading 1`] = `
<div
  class="sc-bdVaJa hBYGDq"
  data-test="Loading"
>
  <div
    class="sc-bwzfXH gDVoy"
  />
  <div
    class="sc-bwzfXH gDVoy"
  />
  <div
    class="sc-bwzfXH gDVoy"
  />
</div>
`;
